README.md
# Resize images automatically

This extension automatically resizes images to a given maximum height/width right after they have been uploaded to the TYPO3 website.
The aspect ratio is of course kept.

The idea behind this extension is that TYPO3 should make both administrators and editors happy. Administrators want the website’s
footprint on server as small as possible to be able to handle backups efficiently and want the web server to deliver the pages as
quick as possible. On the other hand, editors should be able to do their job and not bother with technical considerations such as
the size of a picture on disk or that uploading their wonderful sunset taken during holidays with their 12 MP camera will slow down
the time rendering of their great photo gallery where pictures are shown with a maximum definition of 800 × 600 pixels. Moreover,
editors are either not aware of this or are simply unable to “prepare” their pictures as they are using a foreign computer (in a
cyber café) or on the road with their laptop, neither of them running their beloved image editing software.

General configuration settings let you choose which directories should be somehow “monitored” for uploaded pictures and define the
file types that should be handled (e.g., “jpg” and “tif” but not “png” nor “gif”) and a file size threshold (e.g., “skip any picture
smaller than 400 KB”). After all, if an editor managed to create a picture of many mega-pixels that weights only a few KB, why
should we bother?


## Screencast

The team from [jweiland.net](https://jweiland.net/) prepared a screencast showing how to install and configure this extension in
your TYPO3 website: https://jweiland.net/video-anleitungen/typo3/interessante-typo3-extensions/image-autoresize.html.


## Screenshot

The following two figures show how an administrator may easily configure rules to resize uploaded images:

![General Configuration][general-configuration]

![General Options][general-options]

[general-configuration]: https://raw.githubusercontent.com/xperseguers/t3ext-image_autoresize/master/Documentation/Images/general-configuration.png "General Configuration"

[general-options]: https://github.com/xperseguers/t3ext-image_autoresize/raw/master/Documentation/Images/general-options.png "General Options"


## Full Documentation

Please head to https://docs.typo3.org/typo3cms/extensions/image_autoresize/ for the complete extension manual.


ChangeLog
2013-06-03  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] OpenOffice manual is obsolete

2013-06-02  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] Convert documentation to ReST

2013-05-14  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] Remove material for TYPO3 < 4.5 LTS

2013-05-02  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.3.2

2013-05-01  Xavier Perseguers  <xavier@causal.ch>

	* [BUGFIX] CSS class "typo3-noDoc" is not applied anymore in TYPO3 6.1

2013-04-02  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.3.1
	* [BUGFIX] #46752: TYPO3 6.0 doesn't save configs (thanks to Sascha Heilmeier)

2013-03-18  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Invalid descriptions in XLIFF
	* [TASK] Include official XLIFF translation files

2013-03-17  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.3.0
	* [TASK] #12892: Reset EXIF Orientation when using GraphicsMagick

2013-03-15  Xavier Perseguers  <xavier@causal.ch>

	* [BUGFIX] #12892: EXIF rotation does not work with GraphicsMagick
	* [BUGFIX] #46341: A few uncommon EXIF rotations are not handled
	* [BUGFIX] #13065: PNG with alpha-channel fails
	* [BUGFIX] #32603: Can't access configuration
	* [BUGFIX] #46326: Only one file is processed
	* [BUGFIX] Incorrect hook registration in TYPO3 < 6.0
	* [BUGFIX] #46280: Hook processData_postProcessAction receives file resources

2013-03-14  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] #46287: Use namespaces
	* [!!!][TASK] #46284: Remove compatibility code for TYPO3 < 4.5 LTS
	* [!!!][TASK] #46283: Drop support for PHP 5.2
	* [BUGFIX] #46275: Configuration is written to localconf.php

2012-09-24  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.2.0
	* [BUGFIX] #9399: Does not work with DAM (thanks to Eventex Nord for sponsoring)

2012-03-18  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.1.1
	* [BUGFIX] #31697: No style is applied on the Backend module in TYPO3 4.5.x

2011-11-07  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.1.0

2011-11-05  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] Update documentation as the configuration wizard moved to a dedicated Backend module
	* [BUGFIX] #30193: PHP Catchable Fatal Error with no watched image types
	* [BUGFIX] #13470: Cannot configure rules with ExtJS-based Extension Manager

2011-08-30  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] #9399: DAM is currently not compatible
	* [BUGFIX] #12892: Does not work with GraphicsMagick
	* [BUGFIX] #28912: New file won't get created if dimensions equal (thanks to Thomas Deinhamer)
	* [TASK] CGL cleanup

2010-10-14  Xavier Perseguers  <xavier@causal.ch>

	* [FEATURE] #10252: Allow resizing to be deactivated with a rule

2010-09-30  Xavier Perseguers  <xavier@causal.ch>

	* [BUGFIX] #9980: No resizing if option 'automatic orientation' is activated and TYPO3 is using GraphicsMagick

2010-09-17  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Add information about incompatibility with kb_filequota in TYPO3 < 4.5.0 (thanks to Katja Lampela)

2010-08-21  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 1.0.0: Extension is now considered stable

2010-07-28  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] Fix inconsistencies when rendered as HTML on TER

2010-07-27  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Release to TER
	* [TASK] Raise version to 0.5.1
	* [BUGFIX] #9057: No file returned by the hook if it does not match any rule set (thanks to Rachel Foucard)
	* [TASK] Release to TER
	* [TASK] Boost up version to 0.5.0

2010-07-25  Xavier Perseguers  <xavier@causal.ch>

	* [TASK][DOC] Add documentation

2010-07-24  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Raise version to 0.2.0
	* [TASK] Add support for converting the file format
	* [TASK] Add support for wildcard in directory name
	* [TASK] Add multi-select for the image types

2010-07-23  Xavier Persegeurs  <xavier@causal.ch>

	* [TASK] Raise version to 0.1.0 (beta)
	* [TASK] Implement check against a list of rule sets
	* [TASK] Add context sensitive help to all fields

2010-07-22  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Add option to keep metadata information

2010-07-21  Xavier Perseguers  <xavier@causal.ch>

	* [TASK] Initial upload


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "image_autoresize".
 *
 * Auto generated 02-05-2013 15:28
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Resize images automatically',
	'description' => 'This extension allows you to automatically resize down huge pictures uploaded by your editors.',
	'category' => 'be',
	'author' => 'Xavier Perseguers',
	'author_email' => 'xavier@causal.ch',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => 'mod1',
	'doNotLoadInFE' => 1,
	'state' => 'stable',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => 'Causal Sàrl',
	'version' => '1.3.3-dev',
	'constraints' => array(
		'depends' => array(
			'php' => '5.3.3-0.0.0',
			'typo3' => '4.5.0-6.2.99',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:24:{s:9:"ChangeLog";s:4:"7b7a";s:12:"ext_icon.gif";s:4:"5cae";s:17:"ext_localconf.php";s:4:"903e";s:14:"ext_tables.php";s:4:"685f";s:31:"Classes/Hook/FileUploadHook.php";s:4:"c3aa";s:34:"Classes/Utility/JpegExifOrient.php";s:4:"fa8e";s:42:"Classes/v4/class.user_fileupload_hooks.php";s:4:"5619";s:36:"Configuration/FlexForms/Rulesets.xml";s:4:"c52d";s:29:"Configuration/TCA/Options.php";s:4:"34a6";s:40:"Resources/Private/Language/locallang.xlf";s:4:"b58a";s:40:"Resources/Private/Language/locallang.xml";s:4:"3940";s:53:"Resources/Private/Language/locallang_csh_flexform.xlf";s:4:"852f";s:53:"Resources/Private/Language/locallang_csh_flexform.xml";s:4:"ab6b";s:63:"Resources/Private/Language/locallang_csh_tx_imageautoresize.xlf";s:4:"a0e9";s:63:"Resources/Private/Language/locallang_csh_tx_imageautoresize.xml";s:4:"edb3";s:44:"Resources/Private/Language/locallang_tca.xlf";s:4:"d976";s:44:"Resources/Private/Language/locallang_tca.xml";s:4:"b9ff";s:14:"doc/manual.sxw";s:4:"f72f";s:13:"mod1/conf.php";s:4:"198a";s:14:"mod1/index.php";s:4:"6d05";s:18:"mod1/locallang.xlf";s:4:"e3e3";s:18:"mod1/locallang.xml";s:4:"26e3";s:22:"mod1/locallang_mod.xlf";s:4:"b0fa";s:22:"mod1/locallang_mod.xml";s:4:"3bb1";}',
	'suggests' => array(
	),
);

?>